DataFormats/HGCalReco/interface/HGCalSoAClustersService.h

```diff
@@ -0,0 +1,19 @@
+#ifndef DataFormats_HGCalReco_interface_HGCalSoAClustersService_h
+#define DataFormats_HGCalReco_interface_HGCalSoAClustersService_h
+
+#include "DataFormats/SoATemplate/interface/SoACommon.h"
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+#include "DataFormats/SoATemplate/interface/SoAView.h"
+
+GENERATE_SOA_LAYOUT(
+    HGCalSoAClustersServiceLayout,
```

Comment:
Could you rename this to something other than `Service` ?

If it is never stored in the `Event`, you can also move it out of `DataFormats` and into the package that uses it.